src/object.rs

```diff
@@ -971,6 +1022,26 @@ impl DirEntry {
     }
 }
 
+// implement setters for DirEntry's metadata fields
+impl DirEntry {
+    /// record the ETag of `DirEntry`'s corresponding object
+    pub(crate) fn set_etag(&mut self, etag: String) {
```

Comment:
```suggestion
    pub(crate) fn set_etag(&mut self, etag: &str) {
```

Keep the same API style.